scipy/stats/tests/test_morestats.py

```diff
@@ -2541,3 +2542,95 @@ def test_result(self, correction):
 
         res = stats.median_test(x, y, correction=correction)
         assert_equal((res.statistic, res.pvalue, res.median, res.table), res)
+
+
+class TestDirectionalFuncs:
+    # Reference implementations are not available
+    def test_directionalmean_correctness(self):
+        # Data from Fisher: Dispersion on a sphere, 1953 and
+        # Mardia and Jupp, Directional Statistics.
+        # For x coordinate, result for spherical mean does not agree with
+        # Mardia & Jupp, probably due to rounding errors in reference, that's
+        # why the first entry of the spherical mean reference is different
+        # than in the book.
+
+        decl = -np.deg2rad(np.array([343.2, 62., 36.9, 27., 359.,
+                                     5.7, 50.4, 357.6, 44.]))
+        incl = -np.deg2rad(np.array([66.1, 68.7, 70.1, 82.1, 79.5,
+                                     73., 69.3, 58.8, 51.4]))
+        data = np.stack((np.cos(incl) * np.cos(decl),
+                         np.cos(incl) * np.sin(decl),
+                         np.sin(incl)),
+                        axis=1)
+
+        directional_mean = stats.directionalmean(data)
+        mean_rounded = np.round(directional_mean, 3)
+
+        reference_mean = np.array([0.298, -0.135, -0.945])
+        assert_allclose(mean_rounded, reference_mean)
+
+    def test_directionalmean_2d(self):
+        # Test that for circular data directionalmean yields
+        # same result as circmean
+        rng = np.random.default_rng(seed=42)
```

Comment:
We don't want to show "basic" seeds anymore
```suggestion
        rng = np.random.default_rng(abs(hash("directionalmean")))
```

Reply:
I'm not sure that there's a guarantee that `hash()` returns the same thing for the same string on all platforms and Python versions. This would probably be a worse example of good seeding practice than `42`.

AFAIC, we can use whatever seeds we like in a test. It's the documentation that we are primarily concerned about. If you do want to show off good seeding practice, just grab a large integer offline and put it in as a literal. E.g. use `secrets.randbits(128)` to get a number and put it in like so:

```python
rng = np.random.default_rng(0xec9a6899d5a2830e0d1af479dbe1fd0c)
```

Reply:
Thank you @rkern for the precisions. I updated the suggestion.

Do you think text could be an input of `np.random.default_rng`? In the end there is more entropy with text and we could express good seed with shorter elements.

Reply:
Nominally, yes, via cryptographic hashing, but I don't think it will help the seed quality situation. People will use a (relatively) small handful of meaningful strings like `'directionalmean'` as you did rather than actually random text. So you're back to the same problem of restricting oneself to only 32-bit numbers, only probably a smaller set. I suspect they will be at least as likely to be copied verbatim from code snippet to code snippet as the jokey numbers like `42`; for that matter, text is much more likely to attract and propagate joke seeds. High-entropy texts (say from a password generator) are likely to trip up security checkers that look for credentials that were accidentally checked in.

Remember, there's nothing wrong with any of these seeds by themselves; there is no low-quality seed per se, there are just low-quality seed-generating processes. `0`, `10`, `42`, `12345` are all perfectly fine by themselves; the hashing inside of `SeedSequence` ensures that the PRNG gets initialized well (avoiding issues like LFSRs' zeroland). The only issue is when they get reused in production statistics code. Allowing text doesn't seem to me to encourage high-quality seed-generating processes.